### PR TITLE
Phpdoc fixup

### PR DESCRIPTION
**Pull request by samuelwilliams**
Addresses formatting issues with some of the PHPDoc Blocks

### CODE REVIEW

**Comment by rlanvin** [comment]
This one is looking good, but I have to wait for your fix on #12 and #13 before I can merge.

File: src/IP.php [hunk 1]
```diff
@@ -6,252 +6,281 @@
...
+    {
+        if ($base < 2 || $base > 36) {
+            throw new \InvalidArgumentException('Base must be between 2 and 36 (included)');
+        }
+
+        $value = gmp_strval($this->ip, $base);
+
+        // fix for newer versions of GMP (> 5.0) in PHP 5.4+ that removes
+        // the leading 0 in base 2
+        if (2 == $base) {
```
Comment (rlanvin) [inline src/IP.php]: No Yoda condition

File: src/IPBlock.php [hunk 2]
```diff
@@ -6,494 +6,546 @@
...
+     * @internal
+     * Check if the prefix is valid
+     *
+     * @param mixed $prefix
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function checkPrefix($prefix)
+    {
+        if ('' === $prefix || null === $prefix || false === $prefix || $prefix < 0 || $prefix > $this->getMaxPrefix()) {
```
Comment (rlanvin) [inline src/IPBlock.php]: No Yoda condition

File: src/IPBlock.php [hunk 3]
```diff
@@ -6,494 +6,546 @@
...
+    /**
+     * Determine if the current block contains an IP address or block.
+     *
+     * @param mixed $ip_or_block
+     *
+     * @return bool
+     */
+    public function contains($ip_or_block)
+    {
+        if ((is_string($ip_or_block) && false !== strpos($ip_or_block, '/')) || $ip_or_block instanceof IPBlock) {
```
Comment (rlanvin) [inline src/IPBlock.php]: No Yoda condition

File: src/IPBlock.php [hunk 4]
```diff
@@ -6,494 +6,546 @@
...
+    }
+
+    /**
+     * Return the number of IP addresses in the block.
+     *
+     * @return string numeric string (can be huge)
+     */
+    public function getNbAddresses()
+    {
+        if (null === $this->nb_addresses) {
```
Comment (rlanvin) [inline src/IPBlock.php]: No Yoda condition

File: src/IPv4.php [hunk 5]
```diff
@@ -6,137 +6,136 @@
...
 
-	public function getVersion()
-	{
-		return self::IP_VERSION;
-	}
+                $this->ip = $ip;
+            } else {
+                throw new \InvalidArgumentException("$ip is not a valid IPv4 address");
+            }
+        } elseif ((is_resource($ip) && 'GMP integer' == get_resource_type($ip)) || $ip instanceof \GMP) {
```
Comment (rlanvin) [inline src/IPv4.php]: No Yoda condition

File: src/IPv4.php [hunk 6]
```diff
@@ -6,137 +6,136 @@
...
+            // float (or double) with an integer value
+            $ip = gmp_init(sprintf('%s', $ip), 10);
+            if (gmp_cmp($ip, 0) < 0 || gmp_cmp($ip, self::MAX_INT) > 0) {
+                throw new \InvalidArgumentException(sprintf('The double %s is not a valid IPv4 address', gmp_strval($ip)));
+            }
+            $this->ip = $ip;
+        } elseif (is_string($ip)) {
+            // binary string
+            if (!ctype_print($ip)) {
+                if (4 != strlen($ip)) {
```
Comment (rlanvin) [inline src/IPv4.php]: No Yoda condition

File: src/IPv4.php [hunk 7]
```diff
@@ -6,137 +6,136 @@
...
+    }
 
-		return $ip;
-	}
+    /**
+     * Return true if the address is reserved per iana-ipv4-special-registry.
+     */
+    public function isPrivate()
+    {
+        if (null === $this->is_private) {
```
Comment (rlanvin) [inline src/IPv4.php]: No Yoda condition

File: src/IPv6.php [hunk 8]
```diff
@@ -6,128 +6,128 @@
...
+            elseif (ctype_digit($ip)) {
+                $ip = gmp_init($ip, 10);
+                if (gmp_cmp($ip, '340282366920938463463374607431768211455') > 0) {
+                    throw new \InvalidArgumentException(sprintf('%s is not a valid decimal IPv6 address', gmp_strval($ip)));
+                }
+                $this->ip = $ip;
+            } else {
+                throw new \InvalidArgumentException("$ip is not a valid IPv6 address");
+            }
+        } elseif ((is_resource($ip) && 'GMP integer' == get_resource_type($ip)) || $ip instanceof \GMP) {
```
Comment (rlanvin) [inline src/IPv6.php]: No Yoda condition

File: src/IPv6.php [hunk 9]
```diff
@@ -6,128 +6,128 @@
...
+    }
 
-		return $ip;
-	}
+    /**
+     * Return true if the address is reserved per iana-ipv6-special-registry.
+     */
+    public function isPrivate()
+    {
+        if (null === $this->is_private) {
```
Comment (rlanvin) [inline src/IPv6.php]: No Yoda condition

File: tests/IPBlockTest.php [hunk 10]
```diff
@@ -1,201 +1,202 @@
...
+        );
+    }
+
+    /**
+     * @dataProvider validOperations
+     */
+    public function testPlusMinus($block, $plus, $minus, $result)
+    {
+        $block = IPBlock::create($block);
+        if (null !== $plus) {
```
Comment (rlanvin) [inline tests/IPBlockTest.php]: No Yoda condition

File: tests/IPBlockTest.php [hunk 11]
```diff
@@ -1,201 +1,202 @@
...
+            $this->assertEquals($result, (string) $block->plus($plus), "$block + $plus = $result");
+            $this->assertEquals((string) $block, (string) IPBlock::create($result)->minus($plus), "$result - $plus = $block");
+        } elseif (null !== $minus) {
```
Comment (rlanvin) [inline tests/IPBlockTest.php]: No Yoda condition

File: tests/IPBlockTest.php [hunk 12]
```diff
@@ -1,201 +1,202 @@
...
+    }
+
+    /**
+     * @dataProvider invalidOperations
+     * @expectedException \OutOfBoundsException
+     */
+    public function testPlusMinusOob($block, $plus, $minus)
+    {
+        $block = IPBlock::create($block);
+        if (null !== $plus) {
```
Comment (rlanvin) [inline tests/IPBlockTest.php]: No Yoda condition

File: tests/IPBlockTest.php [hunk 13]
```diff
@@ -1,201 +1,202 @@
...
+            $block->plus($plus);
+        } elseif (null !== $minus) {
```
Comment (rlanvin) [inline tests/IPBlockTest.php]: No Yoda condition

File: tests/IPTest.php [hunk 14]
```diff
@@ -1,131 +1,132 @@
...
-			array('::1', null, 2)
-		);
-	}
+    /**
+     * @dataProvider validOperations
+     */
+    public function testPlusMinus($ip, $plus, $minus, $result)
+    {
+        $ip = IP::create($ip);
+        if (null !== $plus) {
```
Comment (rlanvin) [inline tests/IPTest.php]: No Yoda condition

File: tests/IPTest.php [hunk 15]
```diff
@@ -1,131 +1,132 @@
...
+            $this->assertEquals($result, (string) $ip->plus($plus), "$ip + $plus = $result");
+            $this->assertEquals((string) $ip, (string) IP::create($result)->minus($plus), "$result - $plus = $ip");
+        } elseif (null !== $minus) {
```
Comment (rlanvin) [inline tests/IPTest.php]: No Yoda condition

File: tests/IPTest.php [hunk 16]
```diff
@@ -1,131 +1,132 @@
...
-}
\ No newline at end of file
+    /**
+     * @dataProvider invalidOperations
+     * @expectedException \OutOfBoundsException
+     */
+    public function testPlusMinusOob($ip, $plus, $minus)
+    {
+        $ip = IP::create($ip);
+        if (null !== $plus) {
```
Comment (rlanvin) [inline tests/IPTest.php]: No Yoda condition

File: tests/IPTest.php [hunk 17]
```diff
@@ -1,131 +1,132 @@
...
+            $ip->plus($plus);
+        } elseif (null !== $minus) {
```
Comment (rlanvin) [inline tests/IPTest.php]: No Yoda condition

**Comment by rlanvin** [comment]
There are still plenty of Yoda conditions.

**Comment by samuelwilliams** [comment]
Sorry about the Yoda conditions. I've added a config file for PHP-CS-Fixer which has addressed all of the issues.

**Comment by rlanvin** [comment]
Looking good now, merged!

**Pull request merged**